test/jasmine/tests/mesh3d_test.js

```diff
@@ -3,55 +3,298 @@ var createGraphDiv = require('../assets/create_graph_div');
+                expect(trace.cmax).toBe(user[2], 'user cmax');
+                expect(fullTrace.cauto).toBe(full[0], 'full cauto');
+                expect(fullTrace.cmin).toBe(full[1], 'full cmin');
+                expect(fullTrace.cmax).toBe(full[2], 'full cmax');
+            }
+
+            Plotly.plot(gd, [{
+                type: 'mesh3d',
+                x: [0, 1, 2, 0],
+                y: [0, 0, 1, 2],
+                z: [0, 2, 0, 1],
+                i: [0, 0, 0, 1],
+                j: [1, 2, 3, 2],
+                k: [2, 3, 1, 3],
+                intensity: [0, 0.33, 0.66, 3]
+            }])
+            .then(function() {
+                _assert([undefined, undefined, undefined], [true, 0, 3]);
+
+                return Plotly.restyle(gd, 'cmin', 0);
+            })
+            .then(function() {
+                _assert([false, 0, undefined], [false, 0, 3]);
+
+                return Plotly.restyle(gd, 'cmax', 10);
+            })
+            .then(function() {
+                _assert([false, 0, 10], [false, 0, 10]);
+
+                return Plotly.restyle(gd, 'cauto', true);
+            })
+            .then(function() {
+                _assert([true, 0, 10], [true, 0, 3]);
+
+                return Plotly.purge(gd);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+    });
+
+    describe('dimension and expected visibility check and cell/position tests', function() {
+        var gd;
+
+        beforeEach(function() {
+            gd = createGraphDiv();
+        });
+
+        afterEach(function() {
+            Plotly.purge(gd);
+            destroyGraphDiv();
+        });
+
+        function assertVisibility(exp, msg) {
+            expect(gd._fullData[0]).not.toBe(undefined, 'no visibility!');
+            expect(gd._fullData[0].visible).toBe(exp, msg);
         }
 
-        Plotly.plot(gd, [{
-            type: 'mesh3d',
-            x: [0, 1, 2, 0],
-            y: [0, 0, 1, 2],
-            z: [0, 2, 0, 1],
-            i: [0, 0, 0, 1],
-            j: [1, 2, 3, 2],
-            k: [2, 3, 1, 3],
-            intensity: [0, 0.33, 0.66, 3]
-        }])
-        .then(function() {
-            _assert([undefined, undefined, undefined], [true, 0, 3]);
-
-            return Plotly.restyle(gd, 'cmin', 0);
-        })
-        .then(function() {
-            _assert([false, 0, undefined], [false, 0, 3]);
-
-            return Plotly.restyle(gd, 'cmax', 10);
-        })
-        .then(function() {
-            _assert([false, 0, 10], [false, 0, 10]);
-
-            return Plotly.restyle(gd, 'cauto', true);
-        })
-        .then(function() {
-            _assert([true, 0, 10], [true, 0, 3]);
-
-            return Plotly.purge(gd);
-        })
-        .catch(failTest)
-        .then(done);
+        function assertPositions(exp, msg) {
+            expect(gd._fullLayout.scene._scene.glplot.objects[0].positions.length !== undefined).toBe(exp, msg);
```

Comment:
Can we test the actual `length` values, not just whether `positions` and `cells` are arrays like you did in your latest commit?

Reply:
Good call. I'll fix that.